routes/api/v1/favorites.js

```diff
@@ -6,6 +6,47 @@ const configuration = require('../../../knexfile')[environment];
 const database = require('knex')(configuration);
 const fetch = require("node-fetch");
 const env = require('dotenv').config()
+const CurrentForecast = require('../../../current_forecast')
+
+async function fetchCoordinates(location) {
+  let coordinates = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${location},co&key=${process.env.GOOGLE_KEY}`)
+    .then(response => response.json())
+    .then(result => result.results[0].geometry.location)
+    .then(coordinate_data => `${coordinate_data.lat},${coordinate_data.lng}`)
+  return coordinates;
+}
+
+async function fetchWeather(location) {
+  let coordinate_data = await fetchCoordinates(location)
+  let weather_data = await fetch(`https://api.darksky.net/forecast/${process.env.DARKSKY_KEY}/${coordinate_data}`)
+    .then(darksky_data => darksky_data.json())
+  return weather_data;
+}
+
+async function favoritesWeather(locations) {
+  return Promise.all(locations.map(async (location) => {
+    let currentWeatherData = await fetchWeather(location)
+    return {
+      name: location,
+      currentWeather: new CurrentForecast(currentWeatherData.currently)
```

Comment:
According to the project specs, I think these keys are supposed to be `location` and `current_weather`. I'm not entirely sure how important that is though.